### ISSUE TITLE
The casting video reload when back from background (iOS)

### ISSUE THREAD

**Issue by Girum** [react-native-google-cast/react-native-google-cast#374]
Hi! I have installed the component yesterday, and so many testes working fine :pray::skin-tone-2:!

But, in iOS (using Xcode 13/iOS 15), when I'm casting the video, and put the app in background, it goes ok. But when I back from background, the component reload and the casting video on TV as well. This behavior doesn't appear on the Android version.

Someone knows how to avoid this?

The basic implementation was:

```js
const client = useRemoteMediaClient()
const sessionManager = GoogleCast.getSessionManager()
 
useEffect(() => {
        if (client && dataContent != null) {
            // Send the media to your Cast device as soon as we connect to a device
            // (though you'll probably want to call this later once user clicks on a video or something)
            client.loadMedia({
              mediaInfo: {
                contentUrl:
                dataContent.video_urls[streamingQuality],
              },
            })

            client.onMediaProgressUpdated((streamPosition) => { 
                //console.log("STREAMING POSITION: ", streamPosition)
             })

             client.onMediaStatusUpdated((mediaStatus) => {
                 //console.log("MEDIA STATUS UPDATE: ", mediaStatus)
             })

             client.onMediaPlaybackStarted((mediaStatus) => {
                 //console.log("onMediaPlaybackStarted: ", mediaStatus)
             })

             client.onMediaPlaybackEnded((mediaStatus) => {
                //console.log("onMediaPlaybackEnded: ", mediaStatus)
             })

             //GoogleCast.showExpandedControls()
          }

          
    },[client])  
```

And the Cast Button to run. The dataContent.video_urls[streamingQuality] is just my video to run.

**Comment by petrbela** [comment]
This might potentially be an issue with how the library works currently, let me think about this.

**Comment by anhkieet** [comment]
@Girum Have you solved this issue yet?

**Comment by jasonKRR** [comment]
@Girum did you solve this problem?

**Comment by petrbela** [comment]
So the problem is that iOS suspends the session when backgrounding the app and resumes it when coming back.

On Android, however, the cast session is managed by the media router framework that runs as part of the OS itself; therefore, backgrounding the app doesn't suspend the session, hence no resuming either.

I'd personally recommend against calling `loadMedia` in a `useEffect` that depends on the `client`/`session` object, because the session can come and go (device temporarily offline, for example) but the media that's playing on the Cast device should not be affected by the client's (mobile device's) session state.

I'm adding an option `ignoreBackgrounding` to `useCastSession` (and `useClient`) that skips updating the session when backgrounding. Give it a try and let me know if you run into some other issues.

**Comment by petrbela** [comment]
Resolved in 4644b3ea

**Comment by petrbela** [comment]
I also just learned that you can disable iOS suspending the session by setting `options.suspendSessionsWhenBackgrounded = false` when initializing the CastContext in `AppDelegate`. Then it would work the same way as Android does.
https://developers.google.com/cast/docs/reference/ios/interface_g_c_k_cast_options#a0edf44953049a7d888847745a387c9dc
